report_templates/entitlements.erb

```diff
@@ -2,17 +2,32 @@
 name: Subscription - Entitlement Report
 snippet: false
 model: ReportTemplate
+template_inputs:
+- name: Days from Now
+  required: false
+  input_type: user
+  description: "Show subscriptions expiring in <= this many days.  Leave blank to show all entitlements."
+  advanced: false
+  value_type: plain
+  options: "7\r\n30\r\n60\r\n90\r\n120\r\n180"
 require:
 - plugin: katello
   version: 3.14.0
 -%>
-<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 
-'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 
-'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest' -%>
-<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]]).each_record do |host| -%>
-<%-   host.pools.each do |pool| -%>
+<%- days_from_now = input('Days from Now').to_i -%>
+<%- should_filter = !input('Days from Now').empty? -%>
+<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest', 'Subscription Ends In' -%>
+<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]], -%>
+<%- search: should_filter && "pools_expiring_in_days = #{days_from_now}"-%>
+<%- ).each_record do |host| -%>
+<%-   if should_filter -%>
```

Comment:
I do not insist however I think it's a good practice to keep logic in app code base and not in the report. So I'd suggest one more thing to avoid the "filter or not" logic. Could you add one more method to host in Katello PR, something like `host.pools_expiring(days)` which would call `host.pools_expiring_in_days(days)` if `days` is present, `host.pools` otherwise? That way you wouldn't even have to define `should_filter` variable here and just call it. It would either return all pools or pools expiring in a given number of days. Bonus points for doing `.to_i` in there too, to avoid `days_from_now` definition here too.

Reply:
@ares Thanks, I ended up updating `pools_expiring_in_days` so that it handles both string and integer arguments.  I kept the `should_filter` variable assignment just for readability, but the logic (and the `.to_i`) is now moved into Katello.